apis/apiextensions/v1/composition_transforms.go

```diff
@@ -60,6 +68,7 @@ type TransformType string
 // Accepted TransformTypes.
 const (
 	TransformTypeMap     TransformType = "map"
+	TransformTypeMatch   TransformType = "match"
```

Comment:
Don't forget to update the CompositionRevision types and conversion code. 😞

Reply:
Is #3315 a solution for this?

Reply:
I added the API changes to CompositionRevision as well so we don't depend on other issues for this.

@negz can you take another look?